vj4/model/adaptor/contest.py

```diff
@@ -49,6 +49,36 @@ def time(jdoc):
           'detail': detail}
 
 
+def _assignment_stat(tdoc, journal):
+  effective = {}
+  for j in journal:
+    if j['pid'] in tdoc['pids'] and not (j['pid'] in effective and effective[j['pid']]['accept']):
+      effective[j['pid']] = j
+
+  def time(jdoc):
+    real = jdoc['rid'].generation_time.replace(tzinfo=None) - tdoc['begin_at']
+    return real.total_seconds()
+
+  def penalty_score(jdoc):
+    score = jdoc['score']
+    exceed_seconds = (jdoc['rid'].generation_time.replace(tzinfo=None) - tdoc['penalty_since']).total_seconds()
```

Comment:
Where does `penalty_since` come from?

Reply:
See ASSDOC for a sample. Its a homework specific attribute.